text/0000-memory_model.md

```diff
@@ -0,0 +1,393 @@
+- Feature Name: N/A
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Giving Rust a memory model. This allows us to understand what exactly is
+undefined behavior, and what is not, in unsafe code.
+
+# Motivation
+[motivation]: #motivation
+
+To allow unsafe code to be written without worrying about whether the compiler
+will miscompile your code.  Our current system is ill-defined, and far too
+cautious (and in other cases, completely undefined; for example, what are the
+semantics of raw pointer aliasing?).
+
+# Detailed design
+[design]: #detailed-design
+
+This is the complicated part :)
+
+## Using a value
+
+Anything which touches a value, and is not either a move or copy from the value,
+move or copy into an lvalue, or taking a reference to the value, is a use of the
+value. Examples of uses are: arithmetic, match, indexing. Examples of things
+which are not uses are: returning, passing a value to a function, `let x = y`,
+and `let x = &y`. A reborrow is not a use.
```

Comment:
A reborrow is not a use of the pointer or the value being pointed to? (best to be clear here, it's probably the latter)